qt/scientific_interfaces/Muon/ALCInterface.cpp

```diff
@@ -285,5 +287,50 @@ void ALCInterface::importPeakData(const std::string &workspaceName) {
   }
 }
 
+void ALCInterface::externalPlotRequested() {
+  MatrixWorkspace_sptr data;
+  std::vector<QHash<QString, QVariant>> kwargs;
+  switch (m_ui.stepView->currentIndex()) {
+  case 0:
+    data = m_dataLoading->exportWorkspace();
+    if (data) {
+      AnalysisDataService::Instance().addOrReplace("ALC_External_Plot_Loaded_Data", data);
+      kwargs.emplace_back(QHash<QString, QVariant>());
+      kwargs[0].insert("marker", QString(".").toLatin1().constData());
+      kwargs[0].insert("linestyle", QString("None").toLatin1().constData());
+      m_externalPlotter->plotSpectra("ALC_External_Plot_Loaded_Data", "0", true, kwargs[0]);
+    }
+    break;
+  case 1: // Spec 0 = Data, Spec 1 = Calc, Spec 2 = Diff
+    data = m_baselineModellingModel->exportWorkspace();
+    if (data) {
+      AnalysisDataService::Instance().addOrReplace("ALC_External_Plot_Baseline_Workspace", data);
+      kwargs.emplace_back(QHash<QString, QVariant>());
+      kwargs[0].insert("marker", QString(".").toLatin1().constData());
+      kwargs[0].insert("linestyle", QString("None").toLatin1().constData());
```

Comment:
The kwargs in kwargs[0] seems to be reused three times. Can we avoid this repeat?